### ISSUE TITLE
nx dependency graph / npm package name mismatch

### ISSUE THREAD

**Issue by IgorAufricht** [nrwl/nx#1843]
## Expected Behavior
When I have an npm package in the monorepo, the dependency graph discovery tooling should find and consider the dependency regardless of the package name / import path specified in `tsconfig.json`.

Secondary issue: When I generate a publishable package with a directory (e.g. `ng g @nrwl/angular:lib lib2 --directory=common --simpleModuleName --publishable`), the name of the npm package and the import path in `tsconfig.json` should be the same.

## Current Behavior
When I have an npm package in the monorepo, the dependency graph discovery tooling doesn't consider the dependency if the package name / import path in `tsconfig.json` doesn't match with the file path of the package.

Secondary issue: When I generate a publishable package with a directory, the name of the npm package (e.g. "@nxrepo/common-lib") and the import path in `tsconfig.json` (e.g. "@nxrepo/common/lib") differ.

### TL;DR
1. Checkout the https://github.com/IgorAufricht/nx-dep-graph-reproduction repo
2. `npm install`
3. `npm run dep-graph`
The dependency graph shows that the `test-app` only depends on lib1 and lib2, but in the main.ts file of the app all 4 libraries are imported:
![dep-graph](https://user-images.githubusercontent.com/13379857/64954254-4ddba100-d885-11e9-9161-9ccaaac5af9b.png)

### The long version
1. `npx create-nx-workspace@8.5.0 nxrepo --preset=angular --cli=angular`
2. `cd nxrepo`
4. `ng g @nrwl/angular:lib lib1 --publishable`
5. `ng g @nrwl/angular:lib lib2 --directory=common --simpleModuleName --publishable`
6. `ng g @nrwl/angular:lib lib3 --directory=common --simpleModuleName --publishable`
7. `ng g @nrwl/angular:lib lib4 --directory=common --simpleModuleName --publishable`
8. Add imports for the first two libraries to the `main.ts` of the app:
```
import { Lib1Module } from '@nxrepo/lib1';
import { Lib2Module } from '@nxrepo/common/lib2';
```

At this point we can already see a problem:
* the `tsconfig.json` path for lib2 is "@nxrepo/common/lib2"
* the package name in `package.json` for lib2 is "@nxrepo/common-lib2" (note the dash instead of a slash)
So the name used to import the package is different whether importing it from the monorepo or from the npm package, which will cause problems when the package is published to npm. To explain further, a simple scenario:
* lib2 peer-depends on lib3
* this means in the monorepo, lib2 contains an import like `import {whatever} from "@nxrepo/common/lib3"`
This works well in the monorepo, but when both libraries are published to npm and consumed outside the monorepo, the import will fail (since the npm package "@nxrepo/common/lib3" doesn't exist, it's called "@nxrepo/common-lib3"). This could be fixed with adding the correct path to the `tsconfig.json` for our external project, but that's not a very nice thing to have to do with npm packages.

Now we tried to fix it by aligning the name of the import with the name of the npm package:
8. Change the import name of lib3 from "@nxrepo/common/lib3" to "@nxrepo/common/lib3" (which is the npm package name). To do this, change the  `tsconfig.json` as follows:
```
"@nxrepo/common/lib2": ["libs/common/lib2/src/index.ts"],
```
9. Add the proper import to the `main.ts`:
```
import { Lib3Module } from '@nxrepo/common-lib3';
```

Now everything seems to work, but there is one problem - the dependency graph doesn't find the dependency:
10. Run `npm run dep-graph`. This correctly includes app's dependency on lib1 and lib2, but doesn't find the dependency on lib3 (I presume because of the import name / file path mismatch).

Another thing we tried was to remove the "common" part altogether:
11. Change the npm package name of lib4 from "@nxrepo/common-lib4" to just "@nxrepo/lib4".
12, Change the import name in `tsconfig.json` accordingly:
```
"@nxrepo/lib4": ["libs/common/lib4/src/index.ts"]
```
13. Add the import to the `main.ts`:
```
import { Lib4Module } from '@nxrepo/lib4';
```
And now we have the same problem as with lib3 - everything seems to work, but the dependency graph doesn't contain the dependency:
14. `npm run dep-graph`

**Comment by IgorAufricht** [comment]
Related issue: #1520

**Comment by pongells** [comment]
Having the same issue - to partially work around this problem we had to switch to a completely flat folder structure for libs (so that package name is the same as tsconfig paths name by default). Not ideal.

**Comment by jdpearce** [comment]
Thanks for raising this @IgorAufricht, and thanks for referencing the other issue you raised. I'm going to mark this as duplicate for now - I believe that fixing the other issue _should_ address this one.

If it doesn't, we can come back to it afterwards.
